### PR TITLE
Gpapp/chromadb

### PR DESCRIPTION
**Pull request by gpapp**
- Description: version check to make sure chromadb >=0.4.0 does not throw an error, and uses the default sqlite persistence engine when the directory is set,
  - Issue: the issue #7887 

For attention of
  - DataLoaders / VectorStores / Retrievers: @rlancemartin, @eyurtsev

### CODE REVIEW

**Comment by baskaryan** [comment]
cc @jeffchuber

**Comment by jeffchuber** [comment]
@baskaryan we are good for this ... we would like to upgrade the default version for new users as in https://github.com/hwchase17/langchain/pull/7749/files however. how do you think we should merge these?

**Comment by jeffchuber** [comment]
@baskaryan our preference would be to migrate users to the new system however. strong opinion loosely held. all new updates will be to ^0.4.0

**Comment by baskaryan** [comment]
> @baskaryan our preference would be to migrate users to the new system however. strong opinion loosely held. all new updates will be to ^0.4.0

down to land this then #7749, that way we get both? new users get latest, old users don't see things break

File: langchain/vectorstores/chroma.py [hunk 1]
```diff
@@ -459,7 +464,12 @@ def persist(self) -> None:
                 "You must specify a persist_directory on"
                 "creation to persist the collection."
             )
-        self._client.persist()
+        import chromadb
+
+        # Maintain backwards compatibility with chromadb < 0.4.0
+        major, minor, _ = chromadb.__version__.split(".")
+        if int(major) >= 0 and int(minor) < 4:
```
Comment (baskaryan) [inline langchain/vectorstores/chroma.py:471]: @gpapp is this supposed to be int(major) == 0?

Comment (gpapp) [reply langchain/vectorstores/chroma.py:471]: Of course it is! Good spot, I managed to miss that in the rush.

Comment (jeffchuber) [reply langchain/vectorstores/chroma.py:471]: good to land this and https://github.com/hwchase17/langchain/pull/7749 once this change lands

**Pull request merged**